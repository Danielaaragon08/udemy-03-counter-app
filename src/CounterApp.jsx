import React, { useState } from "react";
import PropTypes from "prop-types";

export const CounterApp = ({ valorInicialContador }) => {
  const [contador, setContador] = useState(valorInicialContador); // useState es un hook

  const handleIncremntClick = () => {
    setContador(contador + 1);
  };

  const handleDecrement = () => {

    if (contador === 0) return;

    setContador(contador - 1);
    
  };

  const handleReset = () => {
    setContador(valorInicialContador);
  };

  return (
    <>
      <div className="container mt-4">
        <h1>Counter App</h1>
        <hr />
        <h2>
          Contador: <strong>{contador}</strong>
        </h2>
        <button className="btn btn-primary" onClick={handleDecrement}>
          -
        </button>
        <button className="btn btn-primary m-3" onClick={handleReset}>
          Reset
        </button>
        <button className="btn btn-primary" onClick={handleIncremntClick}>
          +
        </button>
      </div>
    </>
  );
};

CounterApp.propTypes = {
  valorInicialContador: PropTypes.number.isRequired,
};

CounterApp.defaultProps = {
  valorInicialContador: 10,
};