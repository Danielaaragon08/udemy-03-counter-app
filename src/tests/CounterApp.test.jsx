import { fireEvent, render, screen } from "@testing-library/react";
import { CounterApp } from "../CounterApp";

describe("Pruebas en <CounterApp>", () => {
  const valorInicialContador = 100;

  test("Debe de hacer match con el snapshot", () => {
    const { container } = render(
      <CounterApp valorInicialContador={valorInicialContador} />
    );

    expect(container).toMatchSnapshot();
  });

  test("Debe de mostrar el valor inicial de 100", () => {
    render(<CounterApp valorInicialContador={valorInicialContador} />);
    expect(screen.getByText(valorInicialContador)).toBeTruthy();

    expect(screen.getByRole("heading", { level: 2 }).innerHTML).toContain(
      "100"
    );
  });

  test("debe de incrementar con el botón +1", () => {
    render(<CounterApp valorInicialContador={10} />);
    fireEvent.click(screen.getByText("+1"));

    expect(screen.getByText("11")).toBeTruthy();
  });

  test("debe de decrementar con el botón -1", () => {
    render(<CounterApp valorInicialContador={10} />);
    fireEvent.click(screen.getByText("-1"));

    expect(screen.getByText("9")).toBeTruthy();
  });

  test("debe de funcionar el bot+on de reset", () => {

    render(<CounterApp valorInicialContador={355} />);
    fireEvent.click(screen.getByText("+1"));
    fireEvent.click(screen.getByText("+1"));
    fireEvent.click(screen.getByText("+1"));

    fireEvent.click(screen.getByRole('button', {name: "btn-reset"}));

    expect(screen.getByText(355)).toBeTruthy();

  });

});
