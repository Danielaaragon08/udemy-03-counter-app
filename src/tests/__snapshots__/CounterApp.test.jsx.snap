// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <CounterApp> Debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="container mt-4"
  >
    <h1>
      Counter App
    </h1>
    <hr />
    <h2>
      Contador: 
      <strong>
        100
      </strong>
    </h2>
    <button
      class="btn btn-primary"
    >
      -1
    </button>
    <button
      aria-label="btn-reset"
      class="btn btn-primary m-3"
    >
      Reset
    </button>
    <button
      class="btn btn-primary"
    >
      +1
    </button>
  </div>
</div>
`;
